src/optimize.jl

```diff
@@ -25,7 +25,40 @@ function CC.finish(interp::XInterpreter, opt::OptimizationState, params::Optimiz
 end
 
 function CC.optimize(interp::XInterpreter, opt::OptimizationState, params::OptimizationParams, @nospecialize(result))
-    ret = @invoke optimize(interp::AbstractInterpreter, opt::OptimizationState, params::OptimizationParams, @nospecialize(result))
+    nargs = Int(opt.nargs) - 1
+    ir = x_run_passes(opt.src, nargs, opt)
+    return finish(interp, opt, params, ir, result)
+end
 
-    return ret
+include("escape_analysis.jl")
+
+# HACK enable copy and paste from Core.Compiler
+function x_run_passes end
+let f() = @eval CC function $(XCompiler).x_run_passes(ci::CodeInfo, nargs::Int, sv::OptimizationState)
+               preserve_coverage = coverage_enabled(sv.mod)
+               ir = convert_to_ircode(ci, copy_exprargs(ci.code), preserve_coverage, nargs, sv)
+               ir = slot2reg(ir, ci, nargs, sv)
```

Comment:
adding `push!(irs, ir)` after this line seems not working, maybe due to the hack?

Reply:
yes Revise won't work for `eval`ed code, restart the REPL in that case.